From fe775b9bcda19c2df39d090997a68b8522310604 Mon Sep 17 00:00:00 2001
From: Andrii Cherniavskyi <achernia@qti.qualcomm.com>
Date: Mon, 19 May 2025 09:29:27 +0200
Subject: [PATCH] gst-omx: dec: Fix potential thread lock/unlock issue.

Don't unlock the stream lock twice if caps negotiation failed: it will
be unlocked after "goto caps_failed"

Signed-off-by: Andrii Cherniavskyi <achernia@qti.qualcomm.com>
---
 omx/gstomxvideodec.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
index 40b9359..5efd25e 100644
--- a/omx/gstomxvideodec.c
+++ b/omx/gstomxvideodec.c
@@ -2181,7 +2181,6 @@ gst_omx_video_dec_loop (GstOMXVideoDec * self)
             port_def.format.video.eColorFormat);
         if (buf)
           gst_omx_port_release_buffer (port, buf);
-        GST_VIDEO_DECODER_STREAM_UNLOCK (self);
         goto caps_failed;
       }
 
-- 
2.49.0

